File: Hello_World_Tutorials/tutorial-source-code/Java_SLL/sllist.java
 /**
 * Created by Lucas Estrella on 1/31/2017.
 * lestrell@uncc.edu
 */
import bridges.connect.Bridges;
import bridges.base.SLelement;

public class sllist {

    public static void main(String[] args) throws Exception {

		final String YOUR_API_KEY = "";
		final String YOUR_USER_ID = "";
		Bridges<String, StudentInfo> bridges = 
			new Bridges<String, StudentInfo>(100, "YOUR-BRIDGES-API-KEY" , 
				"YOUR-BRIDGES-USER-ID");

         					// create the linked list elements with 
							// student data 
		SLelement<StudentInfo> el0 = new SLelement<StudentInfo>( "",
			new StudentInfo(
				"00000000000",
				"Gretel Chaney",
				"CS",
				"g.chaney@generated.com",
				"magenta",
				"blue",
				 9.0,
				 "https://randomuser.me/api/portraits/med/women/45.jpg"
			));
		SLelement<StudentInfo> el1 = new SLelement<StudentInfo>( "",
			new StudentInfo(
				"00000000001",
				"Karol Soderman",
				"SIS",
				"k.soderman@generated.com",
				"magenta",
				"red",
				 11.0,
				"https://randomuser.me/api/portraits/med/women/46.jpg"
			));
		SLelement<StudentInfo> el2 = new SLelement<StudentInfo>( "",
			new StudentInfo(
				"00000000002",
				"Lamont Kyler",
				"BIO",
				"l.kyler@generated.com",
				"yellow",
				"green",
				 12.0,
				"https://randomuser.me/api/portraits/med/men/80.jpg"
			));
		SLelement<StudentInfo> el3 = new SLelement<StudentInfo>( "",
			new StudentInfo(
				"00000000003",
				"Gladys Serino",
				"CS",
				"g.serino@generated.com",
				"blue",
				"magenta",
				 9.0,
				"https://randomuser.me/api/portraits/med/women/2.jpg"
			));
		SLelement<StudentInfo> el4 = new SLelement<StudentInfo>( "",
			new StudentInfo(
				"00000000004",
				"Starr Mcginn",
				"CS",
				"s.mcginn@generated.com",
				"red",
				"yellow",
				 15.0,
				"https://randomuser.me/api/portraits/med/men/87.jpg"
			));

        					//  link the elements
        el0.setNext(el1);
        el1.setNext(el2);
        el2.setNext(el3);
        el3.setNext(el4);

        SLelement<StudentInfo> currentElement = el0;
						// iterate through the list and add visual attributes
						// to the elements
        while(currentElement != null){
						// color the node
            currentElement.getVisualizer().setColor(currentElement.getValue().getFavoriteColor());

            if(currentElement.getNext() != null){

						// color the link
                currentElement
                        .getLinkVisualizer(currentElement.getNext())
                        .setColor(currentElement.getValue().getDislikeColor());
						// adjust link thickness
                currentElement
                        .getLinkVisualizer(currentElement.getNext())
                        .setThickness(currentElement.getValue()
						.getStudentCreditHours() * 0.75);//75 percent thinner
            }

            currentElement.setLabel(currentElement.getValue().getStudentLabel());
            currentElement = currentElement.getNext();
        }

		bridges.setDataStructure(el0);
		bridges.visualize();


    }





}

File: tutorials/testing/c++/sllist.cpp
/**
 * Created by Kalpathi Subramanian, 1/30/18
 * krs@uncc.edu
 */

#include "Bridges.h"
#include "SLelement.h"
#include <string>

using namespace bridges;
using namespace std;


int main (int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
	// set title
	bridges.setTitle("A Single Linked List Example");

	// set  description
	bridges.setDescription("A singly linked list of node with student names, where each node is linked only to the next node in the list. Illustrates Node and link coloring, node size, link thickness and opacity");

	// create the linked list elements with student names
	SLelement<string>  *el0 = new SLelement<string> ("Gretel Chaney", "Gretel Chaney");
	SLelement<string>  *el1 = new SLelement<string> ("Lamont Kyler", "Lamont Kyler");
	SLelement<string>  *el2 = new SLelement<string> ("Gladys Serino", "Gladys Serino");
	SLelement<string>  *el3 = new SLelement<string> ("Karol Soderman", "Karol Soderman");
	SLelement<string>  *el4 = new SLelement<string> ("Starr McGinn", "Starr McGinn");

	//  link the elements
	el0->setNext(el1);
	el1->setNext(el2);
	el2->setNext(el3);
	el3->setNext(el4);

	// SLelement also has an iterator that can be used to iterate over the list
	SLelement_List<string> list(el0);
    
	// print the names of the students in the list using the iterator
    for (auto iter = list.begin(); iter != list.end(); ++iter)
        cout << *iter << endl;

	// set colors for list elements - see the Color class for supported colors
	el0->setColor("red");
	el2->setColor("aliceblue");

	// color the links - must specify a valid terminating element
	el0->getLinkVisualizer(el1)->setColor("green");
	el3->getLinkVisualizer(el4)->setColor("magenta");

	// adjust link thickness
	el0->getLinkVisualizer(el1)->setThickness(2.0f);

	// set node transparency
	el4->setOpacity (0.5f);

	// set node size
	el0->setSize (20);

	// set link label
	el2->getLinkVisualizer(el3)->setLabel("Gladys_to_Karol");

	bridges.setDataStructure(el0);
	bridges.visualize();

	//freeing
	delete el0;
	delete el1;
	delete el2;
	delete el3;
	delete el4;
}
